support/helper.py

```diff
@@ -14,13 +17,190 @@ def keep_feedback(feedback, score, loan_request, validator):
+        row = pd.DataFrame(temp[-1:].values, columns=cols)
+        row.at[0, col] = row.at[0, sum_col]*-1
+        temp = pd.concat([temp, row], axis=0, ignore_index=True)
+        temp.reset_index(drop=True, inplace=True)
+        temp[sum_col] = temp.loc[::-1, col].cumsum()[::-1].shift(-1)
+        temp = temp[:-1]
+        temp[col] = temp[col]*-1
+        data = pd.concat([data, temp], axis=0, ignore_index=True)
+    return data.to_dict('records')
+
+
+def aggregate_dict_by_month(data):
+    df = pd.DataFrame(data).set_index('date')
+    df.index = pd.to_datetime(df.index)
+    df = df.groupby([df.index.year.values, df.index.month.values]).agg({'amount': 'sum', 'limit': 'max'})
+    return df
+
+
+def util_ratio(metadata, data):
+    metadata['credit_card']['util_ratio'] = {}
+    period = [30, 60, 90, 180, 360, 720, 1800]
+    months = [-1, -2, -3, -6, -12, -24, -60]
+    data['util_ratio'] = data['amount'] / data['limit']
+    for p, m in zip(period, months):
+        metadata['credit_card']['util_ratio'][p] = data.iloc[m:].util_ratio.max()
+    return metadata
+
+
+def balances(metadata, lst, key, df=None):
+    metadata[key]['balances'] = {}
+    metadata[key]['accounts'] = {}
+    current, limit, high_balance = [], [], []
+    accounts = list(set([d['account_id'] for d in lst]))
+    metadata[key]['accounts']['total_count'] = len(accounts)
+    for account_id in accounts:
+        temp = [d for d in lst if d['account_id'] == account_id]
+        current.append(temp[-1]['current'])
+        limit.append(temp[-1]['limit'])
+        if key == 'credit_card':
+            df = aggregate_dict_by_month(temp)
+            high_balance.append(df.amount.max())
+        else:
+            high_balance.append(max([d['current'] for d in temp]))
+    metadata[key]['balances']['current'] = current
+    metadata[key]['balances']['limit'] = limit
+    metadata[key]['balances']['high_balance'] = high_balance
+    if df is not None:
+        metadata = util_ratio(metadata, df)
+    return metadata
+
+
+def transactions(metadata, lst, key):
+    ''' regardless how many different account within same account type '''
+    metadata[key]['transactions'] = {}
+    metadata[key]['transactions']['total_count'] = len(lst)
+    metadata[key]['transactions']['timespan'] = lst[0]['timespan']
+    timespan_in_months = metadata[key]['transactions']['timespan'] / 30
+    metadata[key]['transactions']['avg_monthly_count'] = metadata[key]['transactions']['total_count'] / timespan_in_months
+    metadata[key]['transactions']['avg_monthly_cash_flow'] = sum(d['amount']*-1 for d in lst) / timespan_in_months
+    return metadata
+
+
+def late_payment(metadata, lst):
+    metadata['credit_card']['late_payment'] = {}
+    period = [30, 60, 90, 180, 360, 720, 1800]
+    data = [d for d in lst if d['category'] == 'interest']
+    for p in period:
+        metadata['credit_card']['late_payment'][p] = len([d for d in data if d['timespan'] <= p])
+    return metadata
+
```

Comment:
Perhaps we might store the helper functions pertaining to database storage into a `helper_db.py` or `database.py` file.